src/tbbmalloc/tbbmalloc_internal.h

```diff
@@ -339,7 +339,14 @@ class BackRefIdx { // composite index to backreference array
 // Block header is used during block coalescing
 // and must be preserved in used blocks.
 class BlockI {
-    intptr_t     blockState[2];
+    #if __clang__
+        #pragma clang diagnostic push
+        #pragma clang diagnostic ignored "-Wunused-private-field"
+        intptr_t     blockState[2];
+        #pragma clang diagnostic pop // "-Wunused-private-field"
+    #else
+        intptr_t     blockState[2];
+    #endif
```

Comment:
The following code has the same implementation for all compilers:
```suggestion
    #if __clang__
        #pragma clang diagnostic push
        #pragma clang diagnostic ignored "-Wunused-private-field"
    #endif
        intptr_t     blockState[2];
    #if __clang__
        #pragma clang diagnostic pop // "-Wunused-private-field"
    #endif
```